src/services/UserService.js

```diff
@@ -0,0 +1,143 @@
+import {normalizeCity} from "../utils/StringUtils"
+
+
+// Base Services
+export const createUser = (user) =>
+    fetch(`${API_USER_URL}`, {
+        method: 'POST',
+        body: JSON.stringify(user),
+        headers: {
+            'content-type': 'application/json'
+        }
+    })
+        .then(response => response.json())
+
+export const findUserById = (uid) =>
+    fetch(`${API_USER_URL}/${uid}`)
+        .then(response => response.json())
+
+export const findUserByEmail = (email) =>
+    fetch(`${API_USER_URL}/email/${email}`)
+        .then(response => response.json())
+        .catch(
+            err => {
+                // Handle 204 Error: Content not found, return empty JSON
+                return {};
+            })
+
+export const findUserByUsername = (username) =>
+    fetch(`${API_USER_URL}/username/${username}`)
+        .then(response => response.json())
+        .catch(
+            err => {
+                // Handle 204 Error: Content not found, return empty JSON
+                return {};
+            })
+
+export const findAllUsers = async () => {
+    const response = await fetch(`${API_USER_URL}`)
+    return await response.json()
+}
+
+export const updateUser = (uid, user) =>
+    fetch(`${API_USER_URL}/${uid}`, {
+        method: 'PUT',
+        body: JSON.stringify(user),
+        headers: {
+            'content-type': 'application/json'
+        }
+    })
+        .then(response => response.json())
+
+export const deleteUser = async (uid) => {
+    const response = await fetch(`${API_USER_URL}/${uid}`, {
+        method: 'DELETE'
+    })
+    return await response.json()
+}
+
+// Registration Services
+
+const registrationChecks = async (user) => {
+    // Returns a JSON object with a type ('SUCCESS' or 'ERROR')
+    // If type == 'ERROR', will also an error message ('errorMessage')
+    let returnJson = {
+        'type' : 'ERROR'
+    }
+    const emailCheck = await findUserByEmail(user.email);
+    const usernameCheck = await findUserByUsername(user.username)
+
+    // Check to see password and confirm password match
+    if (!checkParamsMatch(user.password, user.confirmPassword)) {
+        returnJson['errorMessage'] = 'Passwords do not match.'
+    }
+
+    // Check to see that all fields are completed
+    else if (!checkForEmptyFields(user)) {
+        returnJson['errorMessage'] = 'All fields of the form must be filled out.'
+    }
+
+    //TODO: Check to see valid email <something@something.com>
+    else if (!checkEmailFormat(user.email)) {
+        returnJson['errorMessage'] = 'Invalid Email.'
+    }
+
+    // Check if email exists already
+    else if (Object.keys(emailCheck).length !== 0) {
+        returnJson['errorMessage'] = 'The Email you have entered is already being used.'
+    }
+
+    // Check if username exists already
+    else if (Object.keys(usernameCheck).length !== 0) {
+        returnJson['errorMessage'] = 'The Username you have entered is already being used.'
+    }
```

Comment:
nit: have you considered creating methods for these aspects of form validation in `FormUtils.js`?

Reply:
As per our discussion, lets keep it in UserService for now, as the method is specific to UserRegistration, rather than a generic method than any service could use.